src/server/HelloCodeaholics.Services/Application/PharmacyService.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using HelloCodeaholics.Core.Domain.Entities;
+using HelloCodeaholics.Core.Interfaces;
+using HelloCodeaholics.Infrastructure;
+using Microsoft.EntityFrameworkCore;
+
+namespace HelloCodeaholics.Services.Application
+{
+    public class PharmacyService : IPharmacyService
+    {
+        private readonly HelloCodeDbContext _context;
+
+        public PharmacyService(HelloCodeDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<Pharmacy?> GetPharmacyById(int id)
+        {
+            return await _context.Pharmacies.Where(x => x.Id == id).FirstOrDefaultAsync();
+        }
+
+        public async Task<List<Pharmacy>> GetPharmacyList()
+        {
+            return await _context.Pharmacies.ToListAsync();
+        }
+
+        public async Task<bool> PharmacyExists(int id)
+        {
+            return await _context.Pharmacies.AnyAsync(x => x.Id == id);
+        }
+        public async Task<Pharmacy> UpdatePharmacy(Pharmacy pharmacy)
+        {
+            pharmacy.UpdatedDate = DateTime.Now;
+            _context.Update(pharmacy);
+            await _context.SaveChangesAsync();
+            return pharmacy;
+        }
+    }
+}
```

Comment:
Couple new C# features that may or may not be on your radar:

[Primary Constructors](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-12.0/primary-constructors)

Simplified namespace declaration: 
![image](https://github.com/shiitake/hello-codeaholics/assets/30686676/48c4d36a-aa37-4555-af54-cef72ec02d42)

Doesn't change anything logically but the code looks a little bit cleaner:

`namespace HelloCodeaholics.Services.Application;

public class PharmacyService(HelloCodeDbContext context) : IPharmacyService
{
    private readonly HelloCodeDbContext _context = context;
    
    public async Task<Pharmacy?> GetPharmacyById(int id)
    {
        return await _context.Pharmacies.Where(x => x.Id == id).FirstOrDefaultAsync();
    }

    public async Task<List<Pharmacy>> GetPharmacyList()
    {
        return await _context.Pharmacies.ToListAsync();
    }

    public async Task<bool> PharmacyExists(int id)
    {
        return await _context.Pharmacies.AnyAsync(x => x.Id == id);
    }
    public async Task<Pharmacy> UpdatePharmacy(Pharmacy pharmacy)
    {
        pharmacy.UpdatedDate = DateTime.Now;
        _context.Update(pharmacy);
        await _context.SaveChangesAsync();
        return pharmacy;
    }
}`

Reply:
I agree.